app/src/main/java/com/codesoom/assignment/repository/TaskRepository.java

```diff
@@ -0,0 +1,60 @@
+package com.codesoom.assignment.repository;
+
+import com.codesoom.assignment.models.Task;
+import com.codesoom.assignment.models.TaskIdGenerator;
+import org.springframework.stereotype.Repository;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Repository
+public class TaskRepository {
+
+    private final TaskIdGenerator taskIdGenerator;
+    private Map<Long, Task> taskMap = new ConcurrentHashMap<>();
+
+    public TaskRepository(TaskIdGenerator taskIdGenerator) {
+        this.taskIdGenerator = taskIdGenerator;
+    }
+
+    public Collection<Task> getTaskList() {
+        return taskMap.values();
+    }
+
+    public Optional<Task> getTaskById(Long id) {
+        return Optional.ofNullable(taskMap.get(id));
+    }
+
+    public Task addTask(Task task) {
+        Task newTask = new Task(taskIdGenerator.nextSequence());
+        newTask.setTitle(task.getTitle());
+        taskMap.put(newTask.getId(), newTask);
+        return newTask;
+    }
+
+    public Optional<Task> replaceTask(Long id, Task task) {
+        Optional<Task> oldTask = Optional.ofNullable(taskMap.get(id));
+        oldTask.ifPresent(it -> {
+            it.setTitle(task.getTitle());
+            taskMap.put(it.getId(), it);
+        });
+        return oldTask;
+    }
+
+    public Optional<Task> updateTask(Long id, Task task) {
+        Optional<Task> oldTask = Optional.ofNullable(taskMap.get(id));
+        oldTask.ifPresent(it -> {
+            it.setTitle(task.getTitle());
+            taskMap.put(it.getId(), it);
+        });
+        return oldTask;
+    }
+
+    public Optional<Task> deleteTask(Long id) {
+        Task task = taskMap.get(id);
+        taskMap.remove(id);
+        return Optional.ofNullable(task);
+    }
```

Comment:
```suggestion
    public Optional<Task> deleteTask(Long id) {
        taskMap.remove(id);
        Task task = taskMap.get(id);
        return Optional.ofNullable(task);
    }
```

이렇게 해도 같긴하네요. `task`를 선언한 곳과 사용하는 곳이 거리가 있어서 헷갈렸어요. 변수를 선언한 곳과 사용하는 곳은 거리가 가까운 것이 좋습니다. 또한 거리 뿐만 아니라 비슷한 것들끼리 뭉쳐있을수록 더 읽기 좋습니다. 우리가 같은 맥락의 글들은 같이 모여있는게 이해하기 쉬운 것처럼요.

위와 같이 변경했더니 `task`가 없어도 될 것 같아요.

```java
    public Optional<Task> deleteTask(Long id) {
        taskMap.remove(id);
        return Optional.ofNullable(taskMap.get(id));
    }
```

Reply:
넵 이전 코드는 의미단위로 뭉쳐있지 않았네요 그리고 인라인으로 리턴하게 수정하였습니다